src/client/Microsoft.Identity.Client/Platforms/Android/AuthenticationContinuationHelper.cs

```diff
@@ -34,20 +35,25 @@ public static void SetAuthenticationContinuationEventArgs(int requestCode, Resul
             logger.Info(string.Format(CultureInfo.InvariantCulture, "Received Activity Result({0})", (int)resultCode));
 
             AuthorizationResult authorizationResult;
-            if (data.Action != null && data.Action.Equals("ReturnFromEmbeddedWebview", StringComparison.OrdinalIgnoreCase))
+
+            if (data != null && data.Action != null && data.Action.Equals("ReturnFromEmbeddedWebview", StringComparison.OrdinalIgnoreCase))
             {
                 authorizationResult = ProcessFromEmbeddedWebview(requestCode, resultCode, data);
             }
-            else if (!String.IsNullOrEmpty(data.GetStringExtra(BrokerConstants.BrokerResultV2)) || requestCode == BrokerConstants.BrokerRequestId) 
+            else if (data != null && (!String.IsNullOrEmpty(data.GetStringExtra(BrokerConstants.BrokerResultV2)) || requestCode == BrokerConstants.BrokerRequestId)) 
```

Comment:
nit: You can simplify the logic here if you remove the `else` clauses and just set the result

Reply:
But we do not want the subsequent if clauses to run if it passes this condition right?

Reply:
Yes, anyway, not important, let's leave it.